src/assignment-3/ERC20Wrapper.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import {ERC20Permit} from "yield-utils-v2/token/ERC20Permit.sol";
+import {IERC20} from "yield-utils-v2/token/IERC20.sol";
+
+/// @title ERC20 wrapping an underlying ERC20 token
+/// @author davidbrai
+/// @notice Users can send a pre-specified ERC20 token and received wrapped tokens in return.
+///     The wrapped tokens can be burned in order to withdraw the deposited tokens.
+contract ERC20Wrapper is ERC20Permit {
+
+    /// @notice The wrapped ERC20 token
+    IERC20 public token;
```

Comment:
Look about using `immutable` here. It's one of the easiest gas optimizations that can be made, and at the same time very effective.

Reply:
Thanks!